Mathlib/RingTheory/PowerSeries/Basic.lean

```diff
@@ -673,6 +673,40 @@ theorem coeff_prod (f : ι → PowerSeries R) (d : ℕ) (s : Finset ι) :
     Equiv.coe_toEmbedding, Finsupp.mapRange.equiv_apply, AddEquiv.coe_toEquiv_symm,
     Finsupp.mapRange_apply, AddEquiv.finsuppUnique_symm]
 
+/-- The `n`-th coefficient of the `k`-th power of a power series.
+$$
+[x^n] P(x)^k \ = \ \sum_{n_1 + n_2 + \cdots + n_k = n} a_{n_1} a_{n_2} \cdots a_{n_k}
+$$
+-/
+lemma coeff_pow (k n : ℕ) (φ : R⟦X⟧) :
```

Comment:
Certainly this does not require that `R` be commutative.

Reply:
It uses `coeff_prod` which itself requires commutativity for the `prod` `BigOperator`:
```
failed to synthesize
  CommMonoid R⟦X⟧
```

Reply:
The remark was about `coeff_pow`.

Reply:
You are right in principle because a product operator in general does not require commutativity. I opened a thread at https://leanprover.zulipchat.com/#narrow/stream/287929-mathlib4/topic/commutativity.20and.20BigOperators